userdata.tpl

```diff
@@ -4,6 +4,7 @@ Content-Type: multipart/mixed; boundary="/:/+++"
 --/:/+++
 Content-Type: text/x-shellscript; charset="us-ascii"
 #!/bin/bash
+set -ex
```

Comment:
I am not a fan of this in userdata scripts because the echo can reveal secrets (echoed data gets logged, logs get shipped to a log aggregator, people without any access to the machines get to read the logs via the log aggregator) and given that users can add snippets to this script without the benefit of context, this risks blowing up a whole node group by something as simple as `grep` failing to find the pattern in the file.

Reply:
Alternatively the userscript could be surrounded in the same way as the powershell userscript surrounds the pre/post. 
-e just ensures that an error raises an error code
-x write to standard error / expand commmand before execution. 

Witout it the shipped log would be quite useless / we don't ship userdata but I expect ymmv.

Reply:
I would leave the `set -ex` up to the user to set. `-x` can expose secrets the script accesses, `-e` doesn't just ensure the error raises an error code, is stops the execution of the script if a command returns a non-zero status. Both `grep` and `diff` routinely return non-zero status in normal operation.